gift_compassion/models/sponsorship_gift.py

```diff
@@ -202,6 +202,10 @@ def _compute_invoice_fields(self):
 
             gift.amount = sum(amounts)
 
+    def _compute_currency(self):
+        # Set gift currency depending on its invoice currency
+        self.currency_id = self.invoice_line_ids.move_id.currency_id
```

Comment:
Will fail if there are multiple related invoice lines
```suggestion
        self.currency_id = self.mapped("invoice_line_ids.move_id.currency_id")[:1]
```